inst/Upgrades.qml

```diff
@@ -0,0 +1,98 @@
+import QtQuick		2.12
+import JASP.Module	1.0
+
+Upgrades
+{
+	Upgrade
+	{
+		functionName:	"bayesianStateSpace"
+		fromVersion:	"0.1.0"
+		toVersion:		"0.16.4"
+
+		ChangeRename	{ from: "factors";										to: "fixedFactors"						}
+		ChangeRename	{ from: "dates";										to: "time"								}
+		ChangeRename	{ from: "postSummaryTable";								to: "posteriorSummaryTable"				}
+		ChangeRename	{ from: "posteriorSummaryCoefCredibleIntervalValue";	to: "posteriorSummaryCiLevel"			}
+		ChangeRename	{ from: "expectedModelSize";							to: "expectedPredictors"				}
+
+		ChangeRename	{ from: "checkboxAr";									to: "autoregressiveComponent"			}
+
+		ChangeJS
+		{
+			name: "lagSelectionMethod"
+			jsFunction: function(options)
+			{
+				switch(options["lagSelectionMethod"])
+				{
+					case "manualAR":		return "manual";
+					case "autoAR":			return "auto";
+				}
+			}
+		}
```

Comment:
This option does not get upgraded, defaults to `manual` regardless of how it was saved. I don't see an issue with the syntax though, perhaps @JorisGoosen would know?

example file: [bsts_0.16.3.jasp.zip](https://github.com/jasp-stats/jaspBsts/files/9268879/bsts_0.16.3.jasp.zip)

Reply:
the syntax is correct, the issue was that [a bit further down](https://github.com/vandenman/jaspBSTS/blob/13c4b25b18e7005f51ee3e0ddca0a5afd34426a0/inst/Upgrades.qml#L85-L88) I overwrite the options with other non-existent options...

@JorisGoosen would it be a lot of work to change [this](https://github.com/jasp-stats/jasp-desktop/blob/8e8c0b02ede6b023654fff4a7862d22eb310d0a5/Desktop/modules/upgrader/upgrade.cpp#L35-L56) into something like this:

```c++
#ifdef JASP_DEBUG
QSet<QString> namesUpdated;
#endif

for(ChangeBase * change : _changes)
		try
		{
			Log::log() << "Checking if condition for change " << change->toString() << " is satisfied, ";
#ifdef JASP_DEBUG
            if (namesUpdated.contains(change->nameToUpdate)
                Log::log() << "Warning! Updater is updating name \"" << change->nameToUpdate "\" twice within one version!";
            else
                namesUpdated.insert(change->nameToUpdate);
#endif

            // other code
```
with some additional call to `namesUpdated.clear()` when the version changes.